File: CoffeBarManagementAngular/src/app/shared/models/stockProducts.ts
import { GetComponentProducts } from './getComponentProducts';
import { ProductComponent } from './productComponent';

export interface StockProducts {
    productId: number;
    name: string;
    unitPrice: number;
    unitMeasure: string;
    availableForUser: boolean;
    complexProduct: boolean;
    categoryId: number;
    quantity: number;
    supplyCheck: number;
    tva: number;
    componentProducts: GetComponentProducts[];
}

File: CoffeBarManagement/CoffeBarManagement/DTOs/Product/StockProducts.cs
namespace CoffeBarManagement.DTOs.Product
{
    public class StockProducts
    {
        public int ProductId { get; set; }
        public string Name { get; set; }
        public double? UnitPrice { get; set; }

        public string UnitMeasure { get; set; }

        public bool? AvailableForUser { get; set; }

        public bool? ComplexProduct { get; set; }

        public int? CategoryId { get; set; }

        public int? Quantity { get; set; }

        public int? SupplyCheck { get; set; }
        public int? TVA { get; set; }

        public List<ComponentProductDto> ComponentProducts { get; set; }
    }
}
